rfcs/72-editor.md

```diff
@@ -0,0 +1,508 @@
+cargo run -p editor
+```
+3. For this reason, the `editor` crate must provide the main function, so the
+   dispatch should ideally have graceful failing if it doesn't exist.
+4. We are using a separate crate so that the end distributable does not need to
+   rely on the editor, developers that _do_ want the deliverable to rely on the editor can call it themselves.
+
+### CLI Interface
+
+1. Simple command line argument parsing can be done with clap.
+2. Simple error handling can be provided with anyhow.
+3. There does not seem to be a standard pretty-output library for rust,
+   prettytable seems to be a start.
+   1. Options like crossterm might also fit the bill, but also seem like overkill.
+4. The manipulator/live viewing components can delegate to the window interface.
+
+### Window Interface
+
+1. Extend from the base crate's world.
+1. Ideally, everything should be do-able through the graphical interface, but
+   some things can be pushed to later releases.
+2. The `bevy_editor_command` window only needs to handle: (1)Credential Storage,
+   (2) Template Downloads, and (3) Project Selection, everything else is handled
+   by the a separate `bevy_editor` window.
+3. The `bevy_editor_command` window must seamlessly transition to the
+   `bevy_editor` window, as if opening a project.
+4. The ui should be made using bevy components.
+
+### New project construction
+
+1. Handled by `bevy_editor_command`.
+2. Retrieve template from asset repo or local template storage.
+3. Build template object mapping template languages to their result.
+    1. For example foo.rs.tera should become foo.rs.
+4. Fill in appropriate template fields from the user's environment.
+5. Add CI workflows.
+6. Add licenses.
+7. Initialize as Git repo.
+
+### Asset management
+
+1. Template management can be done using terra.
+2. Downloading static assets can be managed by reqwest.
+3. Downloading code assets can be done with cargo.
+4. Downloading template assets can be done with libgit2.
+5. Template assets should use `directories` to store system wide.
+
+### Distributable Packaging
+
+1. Bundle `cargo build` with built asset files.
+2. Let developers create per-target custom build scripts.
+3. Should ideally support WASM out of the box.
+
+### Credential/secrets management
+
+1. Should use the operating system's keychain/credential storage for storing
+   secrets.
+2. For now, only for uploading to itch.io in CI builds.
+
+### Scene/Entity manipulator (with possible component editing dispatch).
+
+1. Can use rfc 62 as its data model.
+2. Should provide an editor for adding and removing components to entities.
+3. Should also contain a basic editor for setting the initial values of an
+   entity's components.
+   1. As a bonus, plugins can allow for domain-specific editor for given
+      component types.
+4. The scene viewer should place all objects in a scene at their initial
+   position.
+5. A play button lets users run any given scene with the world's systems.
```

Comment:
This is a major feature: I would like to see this split out into its own section.

Reply:
Would this require defining how states and scenes are bound together?